test/unit/hpa_background_thread.c

```diff
@@ -65,6 +65,19 @@ set_background_thread_enabled(bool enabled) {
 	expect_d_eq(0, err, "Unexpected mallctl failure");
 }
 
+static void
+wait_until_thread_is_enabled(unsigned arena_id) {
+	bool sleeping = false;
+	tsd_t* tsd = tsd_fetch();
+	do {
+		background_thread_info_t *info =
+		    background_thread_info_get(arena_id);
+		malloc_mutex_lock(tsd_tsdn(tsd), &info->mtx);
+		malloc_mutex_unlock(tsd_tsdn(tsd), &info->mtx);
+		sleeping = background_thread_indefinite_sleep(info);
+	} while (!sleeping);
```

Comment:
nit: add a loop count and assert the count is <= 1M or so, to make sure the test doesn't get stuck.